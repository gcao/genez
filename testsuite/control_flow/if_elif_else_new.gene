#!/usr/bin/env gene run

# Test: If-elif-else with new syntax
# Note: There's a known bug where if expressions don't return values correctly for true branches

(fn test_elif []
  # Test 1: Basic elif with score 85
  (var score 85)
  (print "Testing elif with score:" score "-> ")
  (if (>= score 90)
    (then (print "Grade: A"))
    (elif (>= score 80) (print "Grade: B"))
    (elif (>= score 70) (print "Grade: C"))
    (else (print "Grade: F")))
  
  # Test 2: Different scores
  (print "Score: 95 -> ")
  (if (>= 95 90)
    (then (print "Grade: A"))
    (elif (>= 95 80) (print "Grade: B"))
    (elif (>= 95 70) (print "Grade: C"))
    (else (print "Grade: F")))
    
  (print "Score: 75 -> ")
  (if (>= 75 90)
    (then (print "Grade: A"))
    (elif (>= 75 80) (print "Grade: B"))
    (elif (>= 75 70) (print "Grade: C"))
    (else (print "Grade: F")))
    
  (print "Score: 65 -> ")
  (if (>= 65 90)
    (then (print "Grade: A"))
    (elif (>= 65 80) (print "Grade: B"))
    (elif (>= 65 70) (print "Grade: C"))
    (else (print "Grade: F")))
  
  # Test 3: Time-based greeting
  (print "Time: 10 hours -> ")
  (if (< 10 6)
    (then (print "Too early!"))
    (elif (< 10 12) (print "Good morning"))
    (elif (< 10 18) (print "Good afternoon"))
    (elif (< 10 22) (print "Good evening"))
    (else (print "Good night")))
    
  (print "Time: 20 hours -> ")
  (if (< 20 6)
    (then (print "Too early!"))
    (elif (< 20 12) (print "Good morning"))
    (elif (< 20 18) (print "Good afternoon"))
    (elif (< 20 22) (print "Good evening"))
    (else (print "Good night"))))

(test_elif)