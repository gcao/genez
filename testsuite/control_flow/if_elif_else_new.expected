Testing elif with score: 85 -> 
Grade: B
Score: 95 -> 
Grade: A
Score: 75 -> 
Grade: C
Score: 65 -> 
Grade: F
Time: 10 hours -> 
Good morning
Time: 20 hours -> 
Good evening